doc/README.md

```diff
@@ -0,0 +1,421 @@
+# Seedvault App Backup Repository Documentation
+
+This document is a technical description of how Seedvault backs up apps
+installed on an Android device.
+
+It is inspired by the design of the [Restic backup program](https://restic.net/).
+Portions of this document are blatant copies of
+[its documentation](https://restic.readthedocs.io/en/latest/100_references.html).
+However, due to the different nature of Android backups,
+the design [was heavily simplified](#differences-to-restic).
+
+Note that the backup of files is described [in a different document](../storage/doc/design.md).
+
+## Terminology
+
+This section introduces terminology used in this document.
+
+*Repository*: All data produced during a backup is sent to and stored in a repository
+in a structured form.
+
+*Chunk*: Larger files are cut into re-usable chunks that are the unit of de-duplication.
+
+*Blob*: A blob is a chunk stored encrypted and compressed in the repository.
+
+*Snapshot*: A snapshot stands for the state of a collection of apps
+that have been backed up at some point in time.
+The state here means the app data as delivered by the system (may be incomplete or absent)
+and the apps themselves as device specific APK files as installed at time of backup.
+
+*Storage ID*: A storage ID is the SHA-256 hash of the content stored in the repository.
```

Comment:
[Clarity] The storage ID is basically the file name of any file in the repository, right? One that is encrypted so that it is not meaningful to those with access to the Seedvault data files, but lacking the keys? (A mapping of chunk IDs to storage IDs is kept in the snapshot files, which are encrypted.) If that's the case, consider using the term "file name" somewhere in this definition, just to make it all the more obvious. It's "required in order to load the file" because *it is the name of the file*, right?

I know it's described in the Repository Layout section below as well, but it's somewhat more roundabout than necessary. Somehow, it wasn't clear to me that this was the sole purpose of storage IDs (at least, I think that's the case?) until I read lines 326-327 in the Resume Interrupted Backup section.

Reply:
> The storage ID is basically the file name of any file in the repository, right?

No, but it is the beginning of the file name.

>  If that's the case, consider using the term "file name" somewhere in this definition, just to make it all the more obvious.

ok, done.